core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -391,6 +413,48 @@ void setupVersionMetrics() {
              .register(meterRegistry);
     }
 
+    /**
+     * Sets up gauge metric for each server certificate.
+     */
+    private void setupTlsMetrics(SslContext sslContext, String hostname) {
+        final MeterRegistry meterRegistry = config().meterRegistry();
+
+        final SSLSession sslSession = validateSslContext(sslContext);
+        for (Certificate certificate : sslSession.getLocalCertificates()) {
+            try {
+                final X509Certificate x509Certificate = (X509Certificate) certificate;
+                final String commonName = getCommonName(x509Certificate);
+
+                final Supplier<Number> certificateExpirationInfoProvider = () -> {
+                    try {
+                        x509Certificate.checkValidity();
+                    } catch (CertificateExpiredException | CertificateNotYetValidException e) {
+                        return 0;
+                    }
+                    return 1;
+                };
+                Gauge.builder("armeria.server.certificate.validity", certificateExpirationInfoProvider)
```

Comment:
How about using `ToDoubleFunction` that does not create additional `Number` objects.
```java
Gauge.builder("armeria.server.certificate.validity", x509Certificate, ceriticate -> { ... })
```